contributors/design-proposals/node/cache-management.md

```diff
@@ -0,0 +1,220 @@
+resources such as CPU and RAM. **We introduce cache allocation to the arsenal
+of core pinning, RAM allocation, and job priority to reduce interference and
+ensure more predictable performance.** See [whitepaper](https://builders.intel.com/docs/networkbuilders/deterministic_network_functions_virtualization_with_Intel_Resource_Director_Technology.pdf) on the benefits that
+cache allocation provides NFV workloads.
+
+Cache is particularly important to hide memory access latency given the
+disparity in speed of the CPU core (typically measured in GigaHertz) and memory
+access that could be 100s of nanoseconds when there is a last level cache miss.
+Processor cache is typically allocated to an application on a first come, first
+served basis, with cache lines being evicted based on various [cache
+replacement policies](https://en.wikipedia.org/wiki/Cache_replacement_policies) such as Least Recently Used and Least Frequently Used
+among others. Different applications have different [working set](https://en.wikipedia.org/wiki/Working_set) sizes, and some known as 
+streaming applications, constantly pollute the cache with new lines. Consider
+for example a video player or a cryptography application. Without any dedicated
+allocation of cache, streaming applications or those with large working sets,
+for example App[0] running on Core 0 in the figure below, the majority of the
+cache lines may be unavailable to other workloads such as App[1] running on
+Core 1. Further, new workloads may come and go, further contributing to
+performance variability.
+
+![noisy neighbor](https://user-images.githubusercontent.com/3691428/35775410-e2907a88-093c-11e8-8c7f-6cec5f7f3b6f.png)
+
+A “noisy neighbor” on core zero over-utilizes the shared platform cache
+resources causing performance inversion (though the priority app on core one is
+higher priority, it runs slower than expected).
+
+In this initial enablement, given few users have a sense of cache significance
+(other than deployers of  Network Virtual Functions), we are not seeking to
+establish cache as a first class resource, that is a resource that is
+explicitly requested at container launch time. But seeking to control cache
+allocation to a container based on it QoS.
+
+## Proposed Changes
+
+### Extending CPU Manager
+
+The original [cpu manager design proposal](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/cpu-manager.md) alluded to extending its scope
+to control processor cache allocation to obtain more predictable performance. 
+The CPU Manager intercepts all container related actions, in particular add,
+remove, start, and state update. Below we discuss changes for cache allocation. 
+
+### Pod QoS Class
+
+Kubernetes supports a notion of [pod QoS](https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/), the higher the value, the higher
+the priority. We treat the Guaranteed QOS pod as deserving of more cache
```

Comment:
I'm concerned that QoS is the wrong indicator.  It seems completely likely that batch work has limits specified and it seems equally likely (maybe more) that latency-sensitive work might not specify CPU limit (or might set it higher than request).  This latter is the default operating mode for Borg, and I explicitly advise people to not limit CPU unless they have very careful benchmarks and deep understanding of the implications.  It does not work how one would naively expect.

I think we may want some other indicator for this, e.g. an "archetype" selector or similar.  The nice thing about considering this as a new thing is that we can quota it on its own.  Having a limit on # of cores of latency-sensitive work seems useful.

Reply:
A small clarification here, though cache allocation happens alongside CPU pinning done by CPU manager, we do not exactly rely on how CPU manager assigns cores to a workload. We would map cache ways to container process id's. This is a feature through resctrl fs on kernel version >= 4.10